backend/api/EventHandlers/MissionScheduling.cs

```diff
@@ -0,0 +1,343 @@
+                    {
+                        MissionStatus.Ongoing
+                    },
+                    RobotId = robotId,
+                    OrderBy = "DesiredStartTime",
+                    PageSize = 100
+                });
+
+            return ongoingMissions.Any();
+        }
+
+        public async Task FreezeMissionRunQueueForRobot(string robotId)
+        {
+            var robot = (await _robotService.ReadById(robotId))!;
+            robot.MissionQueueFrozen = true;
+            await _robotService.Update(robot);
+            _logger.LogInformation("Mission queue for robot {RobotName} with ID {RobotId} was frozen", robot.Name, robot.Id);
+        }
+
+        public async Task UnfreezeMissionRunQueueForRobot(Robot robot)
+        {
+            robot.MissionQueueFrozen = false;
+            await _robotService.Update(robot);
+            _logger.LogInformation("Mission queue for robot {RobotName} with ID {RobotId} was unfrozen", robot.Name, robot.Id);
+        }
+
+        public async Task UnfreezeMissionRunQueueForRobot(string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                _logger.LogError("Robot with ID: {RobotId} was not found in the database", robotId);
+                return;
+            }
+            await UnfreezeMissionRunQueueForRobot(robot);
+        }
+
+        public async Task StopCurrentMissionRun(string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                _logger.LogError("Robot with ID: {RobotId} was not found in the database", robotId);
+                return;
+            }
+            if (!await OngoingMission(robot.Id))
+            {
+                _logger.LogWarning("Flotilla has no mission running for robot {RobotName} but an attempt to stop will be made regardless", robot.Name);
+            }
+
+            try
+            {
+                await _isarService.StopMission(robot);
+            }
+            catch (HttpRequestException e)
+            {
+                string message = "Error connecting to ISAR while stopping mission";
+                _logger.LogError(e, "{Message}", message);
+                OnIsarUnavailable(robot);
+                throw new MissionException(message, (int)e.StatusCode!);
+            }
+            catch (MissionException e)
+            {
+                string message = "Error while stopping ISAR mission";
+                _logger.LogError(e, "{Message}", message);
+                throw;
+            }
+            catch (JsonException e)
+            {
+                string message = "Error while processing the response from ISAR";
+                _logger.LogError(e, "{Message}", message);
+                throw new MissionException(message, 0);
+            }
+
+            if (robot.CurrentMissionId != null)
```

Comment:
Could we reverse these if statements to guard statements to avoid nested if statements?